catroid/src/test/java/org/catrobat/catroid/test/visualplacement/VisualPlacementPositionCorrectionTest.java

```diff
@@ -0,0 +1,136 @@
+	@InjectMocks
+	VisualPlacementTouchListener listener;
+
+	private final Float initalPositionX = 1.f;
+	private final Float initalPositionY = 1.f;
+	private final Float firstMovementPositionX = 5.f;
+	private final Float firstMovementPositionY = 5.f;
+	private final Float secondMovementPositionX = 6.f;
+	private final Float secondMovementPositionY = 6.f;
+	private final long firstMovementTimestamp = 150;
+	private final long secondMovementTimestamp = 170;
+
+	@Before
+	public void setUp() {
+		PowerMockito.mockStatic(SystemClock.class);
+		when(SystemClock.elapsedRealtime()).thenReturn((long) 0);
+	}
+
+	public void triggerTouchDownEvent(float startPositionX, float startPositionY) {
+		when(motionEvent.getAction()).thenReturn(MotionEvent.ACTION_DOWN);
+		when(imageView.getX()).thenReturn(startPositionX);
+		when(imageView.getY()).thenReturn(startPositionY);
+		listener.onTouch(imageView, motionEvent, coordinateInterface);
+	}
+
+	public void triggerMovementEvent(long timestamp, float positionX, float positionY) {
+		when(motionEvent.getAction()).thenReturn(MotionEvent.ACTION_MOVE);
+		when(motionEvent.getRawX()).thenReturn(positionX);
+		when(motionEvent.getRawY()).thenReturn(positionY);
+
+		when(SystemClock.elapsedRealtime()).thenReturn(timestamp);
+		when(imageView.getX()).thenReturn(positionX);
+		when(imageView.getY()).thenReturn(positionY);
+		listener.onTouch(imageView, motionEvent, coordinateInterface);
+	}
+
+	public void triggerTouchUpEvent() {
+		when(motionEvent.getAction()).thenReturn(MotionEvent.ACTION_UP);
+		listener.onTouch(imageView, motionEvent, coordinateInterface);
+	}
+
+	@Test
+	public void testTouchCorrection() {
+		triggerTouchDownEvent(initalPositionX, initalPositionY);
+		triggerMovementEvent(firstMovementTimestamp, firstMovementPositionX, firstMovementPositionY);
+		triggerMovementEvent(secondMovementTimestamp, secondMovementPositionX, secondMovementPositionY);
+		triggerTouchUpEvent();
+
+		verify(imageView).setX(firstMovementPositionX);
+		verify(imageView).setY(firstMovementPositionY);
+		verify(imageView, times(3)).setX(anyFloat());
+	}
+
+	@Test
+	public void testTouchCorrectionWithInvalidDelay() {
+		long invalidTimestamp = 220;
+
+		triggerTouchDownEvent(initalPositionX, initalPositionY);
+		triggerMovementEvent(firstMovementTimestamp, firstMovementPositionX, firstMovementPositionY);
+		triggerMovementEvent(invalidTimestamp, secondMovementPositionX, firstMovementPositionY);
+		triggerTouchUpEvent();
+
+		verify(imageView, times(2)).setX(anyFloat());
+	}
+
+	@Test
+	public void testTouchCorrectionWithInvalidDistance() {
+		float invalidPositionX = 20.0f;
+		float invalidPositionY = 20.0f;
+
+		triggerTouchDownEvent(initalPositionX, initalPositionY);
+		triggerMovementEvent(firstMovementTimestamp, firstMovementPositionX, firstMovementPositionY);
+		triggerMovementEvent(secondMovementTimestamp, invalidPositionX, invalidPositionY);
+		triggerTouchUpEvent();
+
+		verify(imageView, times(2)).setX(anyFloat());
+	}
+}
```

Comment:
Maybe checking the coordinates would be a better verification. But currently the ones holding the coordinates are mocked. Maybe @84n4n4 has some better ideas about that ?